flutter/macos/Classes/SentryFlutterPlugin.swift

```diff
@@ -0,0 +1,290 @@
+import Cocoa
+import FlutterMacOS
+import Sentry
+import AppKit
+
+public class SentryFlutterPlugin: NSObject, FlutterPlugin {
```

Comment:
This class is basically the same as the iOS one. Except we check for `NSApplication` instead of `UIApplication`. Can we somehow share that code?

Reply:
I believe it could be even the same class, as the Cocoa SDK works for MacOS too https://docs.sentry.io/platforms/apple/guides/macos/ unless there's special things to be done I'm unaware of @philipphofmann knows more

Reply:
yeah, I am pretty sure we can do that somehow. The only difference is `NSApplication.didBecomeActiveNotification` (AppKit) vs `UIApplication.didBecomeActiveNotification` (UIKit) for listening for the start of the application. The rest is identical.

Reply:
https://nshipster.com/swift-system-version-checking/ this seems to be a possibility.

Reply:
Cocoa SDK already does it with compile flags, we can do the same

Reply:
See https://github.com/getsentry/sentry-cocoa/blob/master/Sources/Sentry/SentrySessionTracker.m#L55-L67
we just need to introduce such flags and see how its done in Swift

Reply:
@philipphofmann do you know if it's possible to do that on Swift, such compiler flags? obviously, we can google it but we just wanted to be sure that there are no side effects doing so on Swift.

Reply:
Compiler flags are pretty straightforward in Swift. I use them for example here https://github.com/getsentry/sentry-cocoa/blob/master/Tests/SentryTests/Integrations/SentryOutOfMemoryTrackerTests.swift. I don't think there should be any side effects. There are certain differences exactly like @ueman pointed out, but apart from that, there is not special setup required.

Reply:
I don't see any differences in this class at least.